### DISCUSSION TITLE
How to focus-within and focus-visible at the same time

### DISCUSSION THREAD

**Discussion by paulwongx** [tailwindlabs/tailwindcss#10507] [Help]
I want to focus a `div` surrounding a button but only when keyboard focused. Usually `focus-within` works, but in this case it should only focus on keyboard focus (`focus-visible:`) and not when clicking with the mouse(`focus:`). 

Essentially, I need to combine `focus-within` and `focus-visible`. How can this be done?

Tailwind Play: https://play.tailwindcss.com/ApDB5gSjqv

```html
<div class="flex h-screen items-center justify-center">
  <div class="rounded-lg bg-green-100 px-20 py-20 focus-within:ring-4 focus-within:ring-blue-300">
    <button class="bg-green-200 px-6 py-3 focus:outline-none">Focusable Button</button>
  </div>
</div>
```

Notes:
- Based on [this thread](https://github.com/w3c/csswg-drafts/issues/3080), it looks like w3c doesn't have support for `focus-within-visible`. What's an alternative or round-about way to achieve this?
- It looks like there is support for `:has(:focus)` selector in some browsers but how should this be applied in Tailwind...? [Source](https://github.com/WICG/focus-visible/issues/151)

**Comment by wongjn** [comment]
Indeed `:has(:focus-visible)` seems to meet your needs. You could use an [arbitrary variant](https://tailwindcss.com/docs/adding-custom-styles#arbitrary-variants) to use it Tailwind like [in this example](https://play.tailwindcss.com/IzGX5DyTaD) or if you will be using this often, you could [write a plugin to abstract the variant](https://tailwindcss.com/docs/plugins#adding-variants).

**Reply by paulwongx** [reply]
Wow amazing. Thank you!
Also, in case you wanted to answer on SO since I think this approach is more correct than the current one suggested.
https://stackoverflow.com/questions/75365592/tailwindcss-how-to-focus-within-and-focus-visible-at-the-same-time/75365771?noredirect=1#comment132984315_75365771
